[New Term] Swift Sets: `.symmetricDifference()` (#1299)
Codecademy/docs: content/swiftui/concepts/views/terms/symmetricDifference.md

---

> Title: 'Text'

Reviewer: Typo here, should be '.symmetricDifference()'

Description: 'Returns a new set with the elements.'
Subjects:
  - 'Software Development'
  - 'Mobile Development'
  - 'Computer Science'
Tags:

>   - 'SwiftUI'
>   - 'SwiftUI Views'
>   - 'Buttons'
>   - 'Terms'

Reviewer: ```suggestion
  - 'Sets'
  - 'Methods'
```

CatalogContent:

>   - 'learn-swiftui'
>   - 'paths/build-ios-apps-with-swiftui'

Reviewer: ```suggestion
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
```

---

> A **symmetricDifference** in [SwiftUI](https://www.codecademy.com/resources/docs/swiftui) returns a new set with the elements that are either in the set or in the given

Reviewer: ```suggestion
The **.symmetricDifference()** method returns a new set with the elements that are either in the set or in the given
```

> sequence, but not in both.

Reviewer: Can we move this to the same line as the beginning of the sentence? Line 18

## Syntax

> ```pseudo

Reviewer: ```suggestion

```pseudo
```

> func symmetricDifference<S>(_other: S) -> Set<Element> where Element == S.Element, S:Sequence

Reviewer: I think it's best to show this called on a set `mySet` rather than the way Apple Docs writes it out (it could be confusing to the reader if they're a beginner)

```

> Available when *Element* conforms to *Hashable*

Reviewer: Can we include some examples of `Hashable`s. In other words, what else can this method be called on other than a set.

  

## Example

> The following example adds the text Codecademy is Awesome on the display screen.

Reviewer: ```suggestion

The following example adds the text, `"Codecademy is Awesome"`, on the display screen.
```

```swift
let food: Set = ["Apple", "Banana", "Orange", "Peach"]
let colors = ["Green", "Blue", "Orange", "Peach"]
let eitherFoodOrColor = food.symmetricDifference(colors)
```

> This will display the following:

Reviewer: ```suggestion

This will display the following:

```
